CrimeMonitor/CrimeMonitor.cs

```diff
@@ -0,0 +1,849 @@
+
+                    // Check for discrepancy in logged bond claims
+                    if (total < amount)
+                    {
+                        // Adjust the discrepancy report & remove when zeroed out
+                        FactionReport report = record.factionReports
+                            .FirstOrDefault(r =>  r.crimeDef == Crime.Claim);
+                        if (report != null)
+                        {
+                            report.amount -= Math.Min(amount - total, report.amount);
+                            if (report.amount == 0) { reports.Add(report); }
+                        }
+                    }
+                    // Remove associated bonds claims
+                    record.factionReports = record.factionReports.Except(reports).ToList();
+                }
+                // Adjust the total claims
+                record.claims -= Math.Min(amount, record.claims);
+
+                RemoveRecord(record);
+                update = true;
+            }
+            return update;
+        }
+
+        private void handleBountyAwardedEvent(BountyAwardedEvent @event)
+        {
+            if (@event.timestamp > updateDat)
+            {
+                updateDat = @event.timestamp;
+                _handleBountyAwardedEvent(@event);
+                writeRecord();
+            }
+        }
+
+        private void _handleBountyAwardedEvent(BountyAwardedEvent @event, bool test = false)
+        {
+            // 20% bonus for Arissa Lavigny-Duval 'controlled' and 'exploited' systems
+            StarSystem currentSystem = EDDI.Instance?.CurrentStarSystem;
+            if (currentSystem != null)
+            {
+                currentSystem = LegacyEddpService.SetLegacyData(currentSystem, true, false, false);
+            }
+
+            // Default to 1.0 for unit testing
+            double bonus = (!test && currentSystem?.power == "Arissa Lavigny-Duval") ? 1.2 : 1.0;
+
+            foreach (Reward reward in @event.rewards.ToList())
+            {
+                int shipId = EDDI.Instance?.CurrentShip?.LocalId ?? 0;
+                long amount = Convert.ToInt64(reward.amount * bonus);
+                FactionReport report = new FactionReport(@event.timestamp, true, shipId, Crime.None, currentSystem?.name, amount)
+                {
+                    station = EDDI.Instance?.CurrentStation?.name,
+                    body = EDDI.Instance?.CurrentStellarBody?.name,
+                    victim = @event.faction
+                };
+
+                FactionRecord record = GetRecordWithFaction(reward.faction);
+                if (record == null)
+                {
+                    record = AddRecord(reward.faction);
+                }
+                record.factionReports.Add(report);
+                record.claims += amount;
+            }
+        }
+
+        private void handleBountyRedeemedEvent(BountyRedeemedEvent @event)
+        {
+            if (@event.timestamp > updateDat)
```

Comment:
See above. Move this if statement to the PreHandle method (rather than repeating it for each individual event)?

Reply:
Concur. Done.